Library/ViewModels/DiscoveryViewModel.swift

```diff
@@ -155,17 +162,39 @@ public final class DiscoveryViewModel: DiscoveryViewModelType, DiscoveryViewMode
 
     self.sortsAreEnabled = self.setSortsEnabledProperty.signal.skipNil()
 
-    currentParams
-      .takePairWhen(self.sortPagerSelectedSortProperty.signal.skipNil().skipRepeats(==))
-      .observeValues {
-        AppEnvironment.current.ksrAnalytics.trackDiscoverySelectedSort(nextSort: $1, params: $0)
-      }
+    Signal.combineLatest(
+      currentParams,
+      self.sortPagerSelectedSortProperty.signal.skipNil().skipRepeats(==),
+      prevSortedPage
+    )
+    .filter { $0.1.rawValue != $0.2.rawValue }
+    .observeValues { currentParams, currentSortedPage, prevSortedPage in
+      AppEnvironment
+        .current
+        .ksrAnalytics
+        .trackDiscoverySelectedSort(
+          sort: currentSortedPage,
+          prevSort: prevSortedPage,
+          params: currentParams
+        )
+    }
 
-    currentParams
-      .takePairWhen(swipeToSort)
-      .observeValues {
-        AppEnvironment.current.ksrAnalytics.trackDiscoverySelectedSort(nextSort: $1, params: $0)
-      }
+    Signal.combineLatest(
+      currentParams,
+      swipeToSort,
+      prevSortedPage
+    )
+    .filter { $0.1.rawValue != $0.2.rawValue }
```

Comment:
Since we are reusing this filter logic twice here, probably want to write a private method for it.